### DISCUSSION TITLE
"Zigbee Direct" BLE commissioning Zigbee devices with Zigbee Mesh?

### DISCUSSION THREAD

**Discussion by Hedda** [zigpy/zigpy#1080] [General]
"Zigbee Direct" is apparently a reason for Zigbee + Bluetooth Low Energy concurrent multi-protocol in Zigbee Router devices/firmware?

https://csa-iot.org/newsroom/the-connectivity-standards-alliance-introduces-zigbee-direct-simplifying-integration-with-bluetooth-low-energy-devices/

Zigbee Direct is a protocol standard that enables Bluetooth Low Energy (BLE)-capable devices, like smartphones or tablets, to onboard and optionally control certain Zigbee devices on a network. The Zigbee Direct protocol describes two types of devices: Zigbee Virtual Device (ZVD) and Zigbee Direct Device (ZDD).

A ZVD acts as a full BLE device that communicates with the ZDD over BLE. ZVDs have the option to run a Zigbee stack as well, however, this is not a requirement. In practice, a ZVD could be a smartphone application that acts like a network controller and can commission and interact with other Zigbee Direct Devices on a network.

The other device type, ZDD, is a full Zigbee device running an additional BLE stack that enables BLE communications with the ZVD. A ZDD can be any kind of Zigbee device type including an End Device, Router, or Coordinator / Trust Center. Currently, Silicon Labs only implements the ZDD for EFR32 parts with GSDK 4.3 or higher.

https://csa-iot.org/all-solutions/zigbee/zigbee-direct-faq/

Also, believe requires that devices use a ZVD and/or ZDD enabled firmware that will be part of Zigbee R23 (Zigbee Revision 23) specs?

https://csa-iot.org/wp-content/uploads/2022/12/20-27688-037-zigbee_direct_spec.pdf

- _**Zigbee Direct** can simplify device onboarding (and control) utilizing Bluetooth Low Energy, a technology found in many voice assistant products and mobile devices. Uses new virtual devices **Zigbee Direct Device (ZDD)** and **Zigbee Virtual Device (ZVD)**._
  - https://csa-iot.org/newsroom/the-connectivity-standards-alliance-introduces-zigbee-direct-simplifying-integration-with-bluetooth-low-energy-devices/
    - _**Note!** Starting with Zigbee PRO R23 specification onwards, Zigbee Coordinators are required to implement a minimum set of Zigbee Direct support features, such as handling Zigbee Direct-specific devices and key types. These requirements are extensions to the Zigbee stack as run on top of 802.15.4. Coordinators are NOT required to implement Bluetooth Low Energy._
        - https://csa-iot.org/all-solutions/zigbee/zigbee-direct/
          - https://csa-iot.org/all-solutions/zigbee/zigbee-direct-faq/
            - https://csa-iot.org/newsroom/connectivity-standards-alliances-zigbee-direct-feature-in-development/
            -   https://www.ubisys.de/en/company/ubisysblog-aboutubisys/zigbee-direct-golden-units/
            -   https://www.telink-semi.com/zigbee-direct/
            -   https://www.silabs.com/support/training/zgb-201-what-is-zigbee-direct
            -   https://pt.dsr-corporation.com/news/dsr-supports-zigbee-direct/
              -   https://pt.dsr-corporation.com/news/how-dsr-zigbee-direct-solutions-impact-your-users/
                - https://www.youtube.com/watch?v=TlCX95jTrWQ&ab_channel=DSRCorporation

![image](https://user-images.githubusercontent.com/6320001/196883502-fff3e66a-7b2d-40e6-86bd-1e0f1114ff7f.png)

#### What is Zigbee Direct? 
Zigbee Direct is a new feature of Zigbee that can be added to new or existing Zigbee products to allow your Bluetooth Low Energy devices (e.g., smartphone, tablet, or smart speaker) to be used for commissioning and controlling your Zigbee products.  

This is achieved via connecting a Bluetooth Low Energy device with the Zigbee Direct feature (a smart speaker, phone, etc.), referred to as a Zigbee Virtual Device to a Zigbee device concurrently running Bluetooth Low Energy and Zigbee (a hub, a light bulb, etc.), referred to as a Zigbee Direct Device. This communication channel also carries Zigbee network and application messages through the Zigbee Direct Device to the rest of the Zigbee network.

#### What is a Zigbee Virtual Device (ZVD)? 
A Zigbee Virtual Device (ZVD) is a Bluetooth Low Energy device with the Zigbee Direct Feature performing a certain role (e.g. Trust Center, commissioner) or adopting a certain device type on a Zigbee network (e.g. End Device, Router, Coordinator, Trust Center). The software stack of a ZVD may be constituted of the ZVD application, Zigbee Pro network stack, and the client’s implementation of Zigbee Direct defined Bluetooth Low Energy services.

#### What is a Zigbee Direct Device (ZDD)? 
A Zigbee Direct Device is a device capable of concurrently running Zigbee (e.g. End Device, Router, Coordinator, Trust Center) and Bluetooth Low Energy stacks, implementing the server-side of the Zigbee Direct defined Bluetooth Low Energy services that enables the exchange of data between certain Bluetooth-based devices (ZVDs) and a Zigbee device or network amd allow commissioning of Zigbee devices for operation on the Zigbee network.

#### What are the main use cases that Zigbee Direct addresses? 
There are a number of different use cases that center around commissioning and pairing devices in various situations, including: 

Bluetooth Low Energy is quickly becoming a customary method for commissioning devices. With Zigbee Direct, ubiquitous Bluetooth Low Energy in existing smart speakers, controllers, smartphones, tablets, hubs, etc. can now be used to commission and control Zigbee products with ease and flexibility. 
Zigbee Direct can reduce the number of additional hubs required in a home by utilizing Zigbee Direct to commission and control new Zigbee devices onto an existing Zigbee network
For fully distributed Zigbee networks operating without a hub, Zigbee Direct delivers tools for commissioning and control via existing Bluetooth Low Energy products
Why isn’t there a brand mark for Zigbee Direct and how will the end users know this feature is available in my product? 
Zigbee Direct is a new feature of the Zigbee specification and not a full productized version of the specification. It is recommended that product manufacturers include mention of the Zigbee Direct feature on packaging and other marketing materials to let the user know of its benefits and as an additional selling point for a product. 

#### What can I control with Zigbee Direct? 
Connect to and control any Zigbee device or group of devices on the mesh, for example, read device status, adjust scenes, change levels, create groups, etc., including the ability to now use a UI on a Bluetooth Low Energy-enabled smart device in cases where it doesn’t currently exist. 

You can also commission and configure Zigbee devices for operation on a Zigbee network.

#### Is Zigbee Direct secure?
The Zigbee Direct specification has been designed with security in mind, and is on par with the latest Zigbee security standards. 

#### What are the benefits of Zigbee Direct?
Bluetooth Low Energy commissioning is becoming a widely accepted method for adding new devices to an IoT network. Implementing Zigbee Direct provides  another option for commissioning and pairing that is easy for new Zigbee end users to leverage the familiarity of existing Bluetooth Low Energy devices and drive higher adoption. At the same time this new feature provides more flexibility for experienced Zigbee end users, utilizing the convenience of existing Bluetooth Low Energy connected devices to expand their device networks. 

Who benefits from the Zigbee Direct feature and how is it used by consumers/end users?
Ultimately, the main beneficiary of this feature are end users in both the Consumer and Commercial markets. 

Developers, Ecosystem Providers, and Product/Device Manufacturers also benefit from a new option for commissioning and pairing that is less onerous on the end user and opens up more opportunities for device proliferation.

New entrants to the market can focus on developing their applications and devices and not necessarily be required to build additional hardware/hubs. 

#### Is Zigbee Direct backwards compatible?
Zigbee Direct is supported on two classes of existing Zigbee networks, as long as at least one ZDD is added to the network:

Distributed security networks, that is networks without a central Trust Center
Centralized security networks, that is networks with a Trust Center that either use install codes or do not use install codes for joining devices.

#### Is Zigbee Direct a mandatory feature?
Although the feature is optional, future versions of Zigbee Pro will include a minimum set of mandatory extensions that make Zigbee Coordinators aware of Zigbee Direct devices in the network.

**Comment by Hedda** [comment]
Ubisys is one of the companies that's developing such multi-protocol devices/firmware/stack with support for Zigbee Direct via Bluetooth LE.

https://www.ubisys.de/en/products/for-zigbee-product-developers/platform-zigbee-stack/

**Reply by MattWestb** [reply]
If reading https://csa-iot.org/all-solutions/zigbee/zigbee-direct-faq/ is more or less what Philips HUE is doing in there new BT enabled devices.

I see it as one new commissioning method that is replacing Zigbee Touch Link (Philips patent used in ZLL and ZB3) and can adding device to one centralized (TC based) or distributed (Ala ZLL) network and also controlling the network then its one device in the network with full right.

Very likely is being one key to adding Zigbee devices in Matter systems then ZTL is not working good and the ZLL master key is "leaked" on the internet = not so safe but is OK for my.

**Reply by JohnConnett** [reply]
Are there any readily available tools for **Zigbee Direct**? I have found a [ZRemote Demo Video](https://www.youtube.com/watch?v=TlCX95jTrWQ) but don't have details of its availability.

I have a [Philips Hue Secure Flood Light Camera](https://www.philips-hue.com/en-gb/p/hue-philips-hue-secure-flood-light-camera-eu/8720169177703) and I am interested to see if it is feasible to commission it into an existing Zigbee network using Zigbee Direct.

In the hardware section of the Home Assistant forum I have asked [Philips Hue Secure Camera without a Hue Bridge?](https://community.home-assistant.io/t/philips-hue-secure-camera-without-a-hue-bridge/678816).

**Reply by JohnConnett** [reply]
Unfortunately, it looks like the [Philips Hue Secure Floodlight Camrea](https://www.philips-hue.com/en-gb/p/hue-philips-hue-secure-flood-light-camera-eu/8720169177703) may not be a **Zigbee Direct** device.

The BLE advertisements emitted during setup do not conform to the _BLE Connectable Advertisement example_ in Appendix A of **Zigbee Direct Specification, Version 1.0**. In particular, instead of the 16-bit UUID of the Zigbee Direct Commissioning Service (0xFFF7) it has the 16-bit UUID of Signify Netherlands B.V. (formerly Philips Lighting B.V.) (0xFE0F).

A shame. I was quite looking forward to dabbling with the Zigbee Direct crypto stuff.

**Comment by Hedda** [comment]
> I was quite looking forward to dabbling with the Zigbee Direct crypto stuff.

Not sure if there are any Zigbee Direct Device (ZDD) products yet, but Ubisys Zigbee Direct LED Driver LD6 is a Zigbee Virtual Device (ZVD):

https://www.ubisys.de/en/technology/zigbee-direct/

https://www.ubisys.de/en/products/light-and-consumers/driver-ld6/

They now also offer Ubisys BLEND/Z Luminaire Module as a DIY upgrade option to enable Zigbee Direct in you existing products:

https://www.ubisys.de/en/products/for-zigbee-product-developers/zigbee-luminaire-module/

https://www.ubisys.de/en/products/for-zigbee-product-developers/platform-zigbee-stack/

Ubisys Zigbee Direct LED Driver LD6 demo was based on Qorvo QPG7015M (and they also mention the QPG7015MDK development kit):

https://www.ubisys.de/en/company/ubisysblog-aboutubisys/zigbee-direct-golden-units/

https://www.qorvo.com/products/p/QPG7015M

> Are there any readily available tools for **Zigbee Direct**? 

Not sure, a quick search in Silicon Labs community it sounds as if Ubisys app + Gateway G1 supposedly work with all Zigbee Direct devices:

https://community.silabs.com/s/question/0D58Y0000ASPoi9SQD/zigbee-direct-android-or-ios-app

https://www.ubisys.de/en/solutions/smart-home-app/

https://play.google.com/store/apps/details?id=de.ubisys.smarthome

https://apps.apple.com/us/app/ubisys-smart-home/id686548420

Are you looking to develop your own implementation inside Home Assistant Companion apps if wanted to use [ZHA integration](https://www.home-assistant.io/integrations/zha) in it or?

https://companion.home-assistant.io/

https://github.com/home-assistant/android

https://github.com/home-assistant/iOS

It seems like there is now updated documentation about Zigbee Direct from Zigbee stack SDK manufacturers, such as example Silicon Labs:

https://docs.silabs.com/zigbee/latest/zigbee-direct/

https://www.silabs.com/support/training/zgb-201-what-is-zigbee-direct

https://www.silabs.com/documents/public/presentations/zgb-201-what-is-zigbee-direct.pdf

https://www.youtube.com/watch?v=b7jqB8m0s-I

Note that must use Silicon Labs Gecko SDK 4.3 (GSDK with Zigbee EmberZNet SDK 7.3.0,0) or later to use Zigbee Direct Device (ZDD):

https://github.com/SiliconLabs/gecko_sdk/releases/tag/v4.3.0

  **_Zigbee Direct Device (ZDD) support for:_**
    - **_Onboarding/commissioning_** (Set up or add devices to the Zigbee network)
    - **_Communication to all Zigbee devices without a hub using Bluetooth LE_** (Alpha stage support to send/receive Zigbee data and communicate to Zigbee Direct Device using Bluetooth Low Energy with BLE direct control from a mobile phone)

PS: If you looking to implement in [ZHA integration](https://www.home-assistant.io/integrations/zha) then might be a good idea to check out the companion app commissioning for Matter:

https://www.home-assistant.io/integrations/matter/#bluetooth-used-during-commissioning



https://community.home-assistant.io/t/helpful-hints/655969

Matter developers at Nabu Casa are looking to solve similar things for Bluetooth commissioning of Matter devices with companion apps as currently it only works with Matter based WiFi-devices and it depends on the existing frameworks from Google Android and Apple iOS:

https://www.home-assistant.io/blog/2024/01/25/matter-livestream-blog/

"**_This is what we will be focusing on next:_**

* **_A simpler flow to add Matter devices (a process called commissioning), where we guide our users step-by-step all the way from unboxing a device to being able to control it in Home Assistant. For now, we are focusing on the stable scenarios we mentioned previously. This means that we are improving the commissioning flow in our companion apps, relying on the Matter functionality built into your phone, and using (if necessary) Thread networks managed by the vendor of your phone (Google or Apple)._**

* **_A better handling of sharing of devices between controllers (a functionality called multi-admin). Adding a brand new Matter device (commissioning) and sharing a device that is already controlled by a Matter controller (multi-admin) are not part of the same flow. This difference is dictated by the Matter standard and its security features; a current controller (admin) has to allow a device to join another controller. This is often misunderstood today, and we are trying to create a better flow that guides our users toward the right path for their devices._**

* **_We want to implement some missing features in the current platforms, such as light transitions and scenes._**"

**_This is what we want to achieve in the long run:_**

* **_The ability to use the Home Assistant SkyConnect or Home Assistant Yellow as a Thread border router to connect to Thread-based Matter devices instead of using Apple or Google Border routers._**

**Comment by JohnConnett** [comment]
> Not sure if there are any Zigbee Direct Device (ZDD) products yet, but Ubisys Zigbee Direct LED Driver LD6 is a Zigbee Virtual Device (ZVD):

I might have misunderstood but I think the [ubisys LED-Controller LD6](https://www.ubisys.de/en/products/light-and-consumers/driver-ld6/) _is_ a Zigbee Direct Device (ZDD), capable of communicating with both BLE and Zigbee. The [ubisys Smart Home App](https://www.ubisys.de/en/solutions/smart-home-app/) is a Zigbee Virtual Device (ZVD) only capable of communicating using BLE (to a ZDD).

[ZBOSS](https://dsr-zboss.com) includes _Zigbee® Direct Features_ and is available on a range of platforms.
